pkg/buildkit/buildkitutil.go

```diff
@@ -97,3 +103,47 @@ func buildKitFile(dir, inputfile string) (string, string, error) {
 func GetRemoteBuildkitHostAddress(uid int) string {
 	return "unix://" + filepath.Join("/run/user", fmt.Sprintf("%d", uid), "buildkit/buildkitd.sock")
 }
+
+// GetBuildkitWorkerType returns the type of buildkit worker (oci or containerd).
+func GetBuildkitWorkerType(buildctlBinary, buildkitHost string, shellExecutor executor.ShellExecutor) (string, error) {
+	const (
+		buildkitWorkerExecutorLabelKey = "org.mobyproject.buildkit.worker.executor"
+	)
+	args := buildctlBaseArgs(buildkitHost)
+	args = append(args, "debug", "workers", "--format={{json .}}")
+
+	out, err := shellExecutor.Execute(buildctlBinary, args...)
+	if err != nil {
+		return "", err
+	}
+
+	var workers []map[string]interface{}
+	if err := json.Unmarshal([]byte(out), &workers); err != nil {
+		return "", fmt.Errorf("failed to parse buildkit workers output: %w", err)
+	}
+
+	if len(workers) == 0 {
+		return "", fmt.Errorf("no buildkit workers found")
+	}
+
+	//nolint:lll
+	// Extract the worker type from the first worker, as BuildKit can be configured to use a single worker (either oci or containerd).
```

Comment:
Why not split this comment in two lines? 🙂

Reply:
good catch